### PR TITLE
Add GeoDomain to DNS Nameserver in JSONv5 

### PR DESCRIPTION
**Pull request by karelrooted**
Add geoDomain field for DNS's Nameserver in JSONv5 configuration file. This feature was  available in the JSONv4 configuration format.

This pull request:

- Restore the ability to use geoDomain  in DNS's Nameserver

Usage:

```
{
    "address": {
        "address": "8.8.8.8",
        "port": 53
    },
    "geoDomain": [
        {
            "code": "netflix"
        }
    ]
}
```

### CODE REVIEW

File: app/dns/config.proto [hunk 1]
```diff
@@ -40,6 +40,7 @@ message NameServer {
   optional QueryStrategy query_strategy = 8;
   optional CacheStrategy cache_strategy = 9;
   optional FallbackStrategy fallback_strategy = 10;
+  repeated v2ray.core.app.router.routercommon.GeoSite geo_domain = 68001;
```
Comment (xiaokangwang) [inline app/dns/config.proto:43]: You have added a unused field. Please remove or justify this change.

**Review by xiaokangwang — COMMENTED** [review]
I think it is in general fine. I just have one comment about a unused field in the change.

File: app/dns/config.proto (hunk 1 above)
Comment (karelrooted) [reply app/dns/config.proto]: the unused field is removed, it's added by mistake (I didn't dig deep into the code, just copy the structure from geoip , and added geodomain field accordingly ), the updated commit is squashed

**Comment by xiaokangwang** [comment]
This pull request have been merged. Please send a pull request to the document if you are willing and able.

**Pull request merged**